src/Orchard/Security/Providers/FormsAuthenticationService.cs

```diff
@@ -48,9 +53,21 @@ public class FormsAuthenticationService : IAuthenticationService {
         public void SignIn(IUser user, bool createPersistentCookie) {
             var now = _clock.UtcNow.ToLocalTime();
 
-            // The cookie user data is "{userName.Base64};{tenant}".
-            // The username is encoded to Base64 to prevent collisions with the ';' seprarator.
-            var userData = String.Concat(user.UserName.ToBase64(), ";", _settings.Name);
+            // The cookie user data is "{userName.Base64};{providerKey|providerValue};{providerKey|providerValue}....".
+            // The UserName is encoded to Base64 to prevent collisions with the ';' and '|' seprarators.

```

Comment:
Perhaps also clarify in the comment that the | separator is used to prevent collision with Base64 alphabet.